test/integration/infrastructure/infrastructure_test.go

```diff
@@ -149,61 +150,143 @@ var _ = Describe("Infrastructure tests", func() {
 	})
 
 	Context("with infrastructure that requests existing vpc", func() {
-		var (
-			identifiers infrastructureIdentifiers
-		)
+		It("should successfully create and delete", func() {
+			identifiers := prepareVPC(ctx, clientFactory, *region, vpcCIDR, natGatewayCIDR)
+			defer func() {
+				cleanupVPC(ctx, clientFactory, identifiers)
+			}()
 
-		BeforeEach(func() {
-			identifiers = prepareVPC(ctx, clientFactory, *region, vpcCIDR, natGatewayCIDR)
-		})
+			providerConfig := newProviderConfig(&alicloudv1alpha1.VPC{
+				ID: identifiers.vpcID,
+			}, availabilityZone)
 
-		AfterEach(func() {
-			framework.RunCleanupActions()
-			cleanupVPC(ctx, clientFactory, identifiers)
+			err := runTest(ctx, logger, c, providerConfig, decoder, clientFactory)
+			Expect(err).NotTo(HaveOccurred())
 		})
+	})
 
-		It("should successfully create and delete", func() {
+	Context("with invalid credentials", func() {
```

Comment:
I don't understand why we need this new test case. It doesn't improve residence of existing infrastructure testing

Reply:
As described in the PR description:
> This PR tests https://github.com/gardener/terraformer/pull/93 and https://github.com/gardener/gardener/pull/3950 by adding an integration test case for invalid credentials.
It creates an infrastructure resource but uses invalid credentials and ensures that the ERR_INFRA_UNAUTHENTICATED is correctly detected from the termination message and that the infrastructure can still be deleted afterwards (as it's a no-op).

This is added to capture another case that might happen during reconciliation.